File: tests/infoboxes/infobox2Test.php
<?php



use PHPUnit\Framework\TestCase;
use function WpRefs\Infobox2\do_comments;
use function WpRefs\Infobox2\make_tempse;
use function WpRefs\Infobox2\expend_new;

class infobox2Test extends TestCase
{
    public function test_expend_new_FileText()
    {
        $text_input   = file_get_contents(__DIR__ . "/texts_infobox2/infobox2_input.txt");
        $text_output  = file_get_contents(__DIR__ . "/texts_infobox2/infobox2_output.txt");
        $file_3  = __DIR__ . "/texts_infobox2/infobox2_fixed.txt";
        // --
        $result = expend_new($text_input);
        // --
        $result = preg_replace("/\r\n/", "\n", $result);
        $text_output = preg_replace("/\r\n/", "\n", $text_output);
        // --
        file_put_contents($file_3, $result);
        // --
        $this->assertEquals(trim($text_output), trim($result), "Unexpected result");
    }
    public function test_make_tempse_FileText()
    {
        $text_input   = file_get_contents(__DIR__ . "/texts_infobox2/infobox2_tempse_input.txt");
        $text_output  = json_decode(file_get_contents(__DIR__ . "/texts_infobox2/infobox2_tempse_output.json"), true);
        // --
        $file_3  = __DIR__ . "/texts_infobox2/infobox2_tempse_fixed.json";
        // --
        $result = make_tempse($text_input);
        // --
        file_put_contents($file_3, json_encode($result, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE));
        // --
        $this->assertEquals($text_output, $result, "Unexpected result");
    }
}

File: tests/infoboxes/infobox2Test.py
from __future__ import annotations

import json
from pathlib import Path

from src.infoboxes.infobox2 import expend_new, make_tempse


FIXTURES_DIR = Path(__file__).parent / "texts_infobox2"


class infobox2Test:
    def test_expend_new_FileText(self) -> None:
        text_input = (FIXTURES_DIR / "infobox2_input.txt").read_text(encoding="utf-8")
        text_output = (FIXTURES_DIR / "infobox2_output.txt").read_text(encoding="utf-8")
        result = expend_new(text_input)
        (FIXTURES_DIR / "infobox2_fixed.txt").write_text(result, encoding="utf-8")
        assert text_output.replace("\r\n", "\n").strip() == result.replace("\r\n", "\n").strip()

    def test_make_tempse_FileText(self) -> None:
        text_input = (FIXTURES_DIR / "infobox2_tempse_input.txt").read_text(encoding="utf-8")
        text_output = json.loads((FIXTURES_DIR / "infobox2_tempse_output.json").read_text(encoding="utf-8"))
        result = make_tempse(text_input)
        (FIXTURES_DIR / "infobox2_tempse_fixed.json").write_text(
            json.dumps(result, ensure_ascii=False, indent=4),
            encoding="utf-8",
        )
        assert result == text_output
